test/e2e/multiple_worker_node_groups_test.go

```diff
@@ -89,17 +89,26 @@ func TestDockerKubernetes124UpgradeAndRemoveWorkerNodeGroupsAPI(t *testing.T) {
 			api.WithControlPlaneCount(1),
 			api.RemoveAllWorkerNodeGroups(), // This gives us a blank slate
 		),
-		provider.WithWorkerNodeGroup(framework.WithWorkerNodeGroup("worker-1", api.WithCount(2))),
+		provider.WithWorkerNodeGroup(
+			framework.WithWorkerNodeGroup(
+				"worker-1",
+				api.WithCount(2),
+				api.WithLabel("tier", "frontend"),
```

Comment:
Why are we adding labels and taints to this test?

Reply:
We thought that it might be a good idea to increase coverage without increasing resources for this particular case upgrading a self managed cluster with the api.

I think I need to decouple the upgrades though. Shouldn't be scaling the node group and adding a labels and taints to one worker.